src/ResultSet.php

```diff
@@ -137,10 +160,46 @@ protected function advance()
         }
     }
 
-    protected function yieldTraversable(Traversable $traversable)
+    /**
+     * @param Traversable<int, mixed> $traversable
+     * @return Generator
+     */
+    protected function yieldTraversable(Traversable $traversable): Generator
     {
         foreach ($traversable as $key => $value) {
             yield $key => $value;
         }
     }
+
+    /**
+     * Sets the amount of items a page should contain (only needed for pagination)
+     *
+     * @param ?int $size
+     * @return void
+     */
+    public function setPageSize(?int $size): void
+    {
+        $this->pageSize = $size;
+    }
+
+    /**
+     * Returns the current page calculated from the {@see ResultSet::$offset} and the {@see ResultSet::$pageSize}
+     *
+     * @return int
+     * @throws BadMethodCallException if no {@see ResultSet::$pageSize} has been provided
+     */
+    protected function getCurrentPage(): int
```

Comment:
This should work with the position. The page number calculated by the position should be increased if there's an offset.

Reply:
Not sure how I missed that one... The calculation should now be fixed